joeywellman/endoREG
# Main Models, Views, and Controllers Functional for Application Base

# Description

## User Authentication 
User can Register, Login and Logout of an account (multiple users).


## NavBar Rendering
Given the user is authenticated,
Then the user should see links for 'Home', 'Compliance Plans', 'Evaluation History', 'Resources', 'Profile', and 'Logout' affordances (etc.).

Note: Slight alteration to content.
![image](https://user-images.githubusercontent.com/45610988/61057634-b3ae3700-a3c3-11e9-9b32-b99be108dcc2.png)


## Compliance Plans Page
When the user clicks on the 'Compliance Plans' link in the menu bar,
Then the user will be presented with a list of compliance plans and a button affordance to add a new compliance plan.

Given the user 'clicks' the add a compliance plan button,
Then the user will be presented with a form to populate compliance plan title and description.

Given the user has filled out all fields of the compliance plan form with valid information,
When the user clicks the submit plan button,
Then the user will be redirected to the compliance plan detail view for that compliance plan,
And be able to review / add action items to the compliance plan.

Given the user has filled in all fields of the compliance plan form with any invalid information,
When the user clicks the submit plan button,
Then the user will be immediately notified with error information for each invalid value.

Given the user targets the "Compliance Plans with Action Items" sub-navbar link,
Then the user should be redirected to the cumulative compliance plan's view page
And the user should be presented with a card view that shows compliance plans with action items organized / sorted by (and under) compliance plan titles (labels),
And the user should have link affordances to 'click' compliance plan title hyperlinks to view details,
And the user should have link affordances to 'click' action item hyperlinks to view details of each action item.


## Action Items
Given the authenticated user has submitted a compliance plan,
Then the user will be able to view that compliance plan on the 'Compliance Plans' page,
And be able to review / add action items to the compliance plan by 'clicking' the add new action item affordance.

Given the user 'clicks' the add new action item button,
Then the user will be presented with a action item form to populate action item criteria (title and description),
And have the option to save ("Submit") the action item.

Given the user has filled out all fields of the action item form with valid information,
When the user clicks the submit action item button,
Then the user will be redirected to the compliance plan detail view for that product,
And be able to review / add action items to the compliance plan.

Given the user has filled in all fields of the action item form with any invalid information,
When the user clicks the submit item button,
Then the user will be immediately notified with error information for each invalid value.


## Evaluation Results
Given the user is authenticated and selects the navbar link for "Evaluation Results",
Then the user should be presented with a graph (pie chart) of incomplete vs. completed items.


### Related Ticket(s)
1. User Authentication
1. NavBar Rendering
1. Compliance Plans Page
1. Action Items
1. Evaluation Results


### Steps to Test endoREG
1. "git add ." and "git commit [-m]" whatever you're working on (on your branch)
2. "git checkout master", to switch to local master branch
3. "git fetch --all" will give you access to the branch in question
4. "git checkout main-MVCs" to switch to this branch (PR review requested)
5. If you do not see the most recent version of branch files, try "git pull origin main-MVCs"
1. In Git Bash, 'cd' to "workplace/endoREG"
1. Start "endoREG-MVC.sln"
1. Go to Packet Manager Console and insert command: "add-migration" (changes made)
1. Go to Packet Manager Console and insert command: "update-database" (changes made)
1. Press F5 to run browser / CTRL+R, A to run tests
1. In browser, 'click' "Login" to reach the sign in page
1. Login as Email: "admin@admin.com"; Password: "Admin8*"
1. In browser, review functionality of items (listed above).

# Checklist:
